liboptv/src/orientation.c

```diff
@@ -141,20 +149,803 @@ double weighted_dumbbell_precision(vec2d** targets, int num_targs, int num_cams,
+                                                        (sigmabeta[16]+sigmabeta[17]));
+    }
+
+
+    /* this part is also irrelevant for liboptv, temporary saved to remind us
+       to check outside the orient() and act in the main program to deal with
+       the multi-plane or other type of calibration
+   */
+    /*
+    par_file = fopen_r ("parameters/examine.par");
+    fscanf (par_file,"%d\n", &dummy);
+    fscanf (par_file,"%d\n", &multi);
+    fclose (par_file);
+    if (dummy==1){
+      examine=4;
+    }
+    else{
+      examine=0;
+    }
+    */
+
+    /* show original images with residual vectors (requires globals)
+    is commented out as irrelevant but kept for the future move to the
+    Python side
+    */
+
+    /*
+    printf ("%d: %5.2f micron, ", nr+1, sigma0*1000);
+    printf("\ntest 1 inside orientation\n");
+    for (i=0; i<n_obs-10; i+=2)
+    {
+      n = pixnr[i/2];
+      intx1 = (int) pix[n].x;
+      inty1 = (int) pix[n].y;
+      intx2 = intx1 + resi[i]*5000;  // arbitrary elongation factor for plot
+      inty2 = inty1 + resi[i+1]*5000;
+    orient_x1[n] = intx1;
+    orient_y1[n] = inty1;
+    orient_x2[n] = intx2;
+    orient_y2[n] = inty2;
+    orient_n = n;
+    }
+
+    */
+
+
+
+    if (stopflag){
+        rotation_matrix(&(cal->ext_par));
+        memcpy(cal_in, cal, sizeof (Calibration));
+    }
+    else {
+        /* restore the saved calibration if not converged */
+        // cal = read_calibration(ori_file, add_file, NULL);
+        printf ("\n|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||");
+        printf (" Orientation does not converge");
+        printf ("\n|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||");
+    }
+
+    free(X);
+    free(P);
+    free(y);
+    free(Xbeta);
+    free(Xh);
+    free(resi);
+
+}
+
+/*  raw_orient() uses manually clicked points to setup the first raw orientation of the
+                camera, setting its external parameters: position and angles.
+    Calibration* cal  -      camera calibration object
+    control_par *cpar -     control parameters
+    int nfix	-	number of points, for raw_orient this means typically 4 points
+                    manually selected by the user
+    vec3d fix[]	-	object point data obtained using read_calblock() function and
+                    represents the known calibration body 3D points for the 4 manually
+                    entered points, see man_ori.par
+    target pix[] -	image coordinates obtained from sortgrid(). The points which are
+                    associated with fix[] have real pointer (i+1), others have -999,
+                    for raw_orient these will be typically pix0[] - manually clicked dots
+    Output:
+    Calibration *cal overwritten with the updated orientation, only 6 external parameters
+                     being updated: x,y,z,omega,phi,kappa
+*/
+
+
+void raw_orient (Calibration* cal, control_par *cpar, int nfix, vec3d fix[], target pix[])
```

Comment:
I do not know how to kill raw_orient, i.e. how to call orient() with a limited set of parameters for 4 points only. does it mean to add many if .. else ... statements in the body of orient() ?